tests/test_api_manager.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+import requests_mock
+from mock import *
+
+from APICallManager import *
+
+key = os.environ['FRED_API_TOKEN_REGULATIONS_GOV']
+base_url = 'https://api.data.gov:443/regulations/v3/documents.json?'
+
+
+@pytest.fixture
+def mock_req():
+    with requests_mock.Mocker() as m:
+        yield m
+
+def set_time():
+    mock_time = Mock()
+    mock_time.return_value = None
+    return mock_time
+
+
+def test_success(mock_req):
+    mock_req.get(add_api_key(base_url), status_code=200, text='{}')
+    assert api_call_manager(add_api_key(base_url)) == {}
+
+
+@patch('time.sleep', set_time())
+def test_retry_calls_Failure(mock_req):
+    mock_req.get(add_api_key(base_url), status_code=304)
+    with pytest.raises(CallFailException):
+        api_call_manager(add_api_key(base_url))
+
+def test_permanent_error(mock_req):
```

Comment:
Is this a permanent error or a call fail exception? The name and the application seem to differ.